pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -227,9 +227,14 @@ func (cb *ClusterBuilder) buildDefaultCluster(name string, discoveryType cluster
 		clusterMode:     DefaultClusterMode,
 		direction:       direction,
 		proxy:           cb.proxy,
-		meshExternal:    meshExternal,
+	}
+	if direction == model.TrafficDirectionInbound {
```

Comment:
can you add a comment about this in the code? Its not obvious why we do this

Reply:
Done.